h11/_util.py

```diff
@@ -97,21 +101,21 @@ def validate(regex, data, msg="malformed data", *format_args):
 #
 # The bonus property is useful if you want to take the return value from
 # next_event() and do some sort of dispatch based on type(event).
-class _SentinelBase(type):
-    def __repr__(self):
+class Sentinel(type):
+    def __repr__(self) -> str:
         return self.__name__
 
 
-def make_sentinel(name):
-    cls = _SentinelBase(name, (_SentinelBase,), {})
-    cls.__class__ = cls
+def make_sentinel(name: str) -> Sentinel:
```

Comment:
how about this?

```python
from typing import TypeVar, Type, Tuple, Any, Dict

_T_Sentinel = TypeVar("_T_Sentinel", bound="Sentinel")


class Sentinel(type):
    def __new__(
        cls: Type[_T_Sentinel],
        name: str,
        bases: Tuple[type, ...],
        namespace: Dict[str, Any],
        **kwds: Any
    ) -> _T_Sentinel:
        assert bases == (Sentinel, )
        v = super().__new__(cls, name, bases, namespace, **kwds)
        v.__class__ = v  # type: ignore
        return v

    def __repr__(self) -> str:
        return self.__name__


class CLIENT(Sentinel, metaclass=Sentinel):
    pass
```

Reply:
Looks great, thanks!

Reply:
@pgjones you marked this as resolved - but I don't think it's reflected in the PR?

Reply:
Was missing in the commit, now added.